sklearn/tree/tests/test_tree.py

```diff
@@ -289,9 +298,9 @@ def test_diabetes_underfit(name, Tree, criterion, max_depth):
         max_features=6, random_state=0
     )
     reg.fit(diabetes.data, diabetes.target)
-    score = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
-    assert score < 60 and score > 0, (
-        f"Failed with {name}, criterion = {criterion} and score = {score}"
+    loss = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
```

Comment:
does it make sense to check the mse when the tree was trained with a poisson criteria? The need for a limit of 4000 seems to indicate that the check is super loose?

Reply:
also nit but score seemed like the correct term here. There's no loss in these trees

Reply:
MSE elicits the expectation, so does Poisson deviance. So yes, it makes sense.
I tried to use mean_poisson_deviance, but that did not change much, i.e. still needs a high `max_loss`. I don't know why.

"score" in scikit-learn means "higher values are better". Error and loss, for me, are often synonyms. I can change into "error"?

Reply:
I'm also skeptical here. I changed to use Poisson deviance as score for `"poisson"` splitter. It's much better now.